src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+        /// the entity.
+        /// </summary>
+        /// <param name="sqlMetadataProvider">SqlMetadaProvider object for the given database</param>
+        /// <param name="entityName">Current entity name</param>
+        /// <param name="higherLevelEntityName">Higher level entity name</param>
+        /// <param name="parameters">Dictionary containing the data ncessary to create a record in the table</param>
+        /// <param name="sourceDefinition">Entity's source definition object</param>
+        /// <param name="isLinkingEntity">Indicates whether the entity is a linking entity</param>
+        /// <returns>Created record in the database as a dictionary</returns>
+        private Dictionary<string, object?> BuildAndExecuteInsertDbQueries(ISqlMetadataProvider sqlMetadataProvider,
+                                                                           string entityName,
+                                                                           string higherLevelEntityName,
+                                                                           IDictionary<string, object?> parameters,
+                                                                           SourceDefinition sourceDefinition,
+                                                                           bool isLinkingEntity)
+        {
+            SqlInsertStructure sqlInsertStructure = new(entityName: entityName,
+                                                        sqlMetadataProvider: sqlMetadataProvider,
+                                                        authorizationResolver: _authorizationResolver,
+                                                        gQLFilterParser: _gQLFilterParser,
+                                                        mutationParams: parameters,
+                                                        httpContext: GetHttpContext(),
+                                                        isLinkingEntity: isLinkingEntity);
+
+            IQueryBuilder queryBuilder = _queryManagerFactory.GetQueryBuilder(sqlMetadataProvider.GetDatabaseType());
+            IQueryExecutor queryExecutor = _queryManagerFactory.GetQueryExecutor(sqlMetadataProvider.GetDatabaseType());
+
+            // When the entity is a linking entity, the higher level entity's name is used to get the
+            // datasource name. Otherwise, the entity's name is used.
+            string dataSourceName = isLinkingEntity ? _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(higherLevelEntityName)
+                                                    : _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(entityName);
+            string queryString = queryBuilder.Build(sqlInsertStructure);
+            Dictionary<string, DbConnectionParam> queryParameters = sqlInsertStructure.Parameters;
+
+            List<string> exposedColumnNames = new();
```

Comment:
Can the method `ISqlMetadataProvider.TryGetExposedFieldToBackingFieldMap` be used here?

Reply:
updated to use `ISqlMetadataProvider.TryGetExposedFieldToBackingFieldMap`